### ISSUE TITLE
Единая схема поиска по местам

### ISSUE THREAD

**Issue by Limych** [Limych/GeniBase#66]
Взаимосвязано с #47.

Одно и то же географическое место могло иметь много разных названий в течение времени. Необходимо выработать единую общую схему нечёткого поиска по местам.

**Comment by a-KaDeMi-k** [comment]
Вообще, в географическом справочнике явно не хватает периода существования данной АТ единицы.
Но если мы его введём, тогда уже старые и новые наименования одного и того же нас. пункта надо заводить двумя записями и ещё делать отдельную таблицу связей между ними.
На данном этапе - это доп. нагрузка на нас и усложнение запросов к базе. Так что пока, думаю, не стоит.

**Comment by Limych** [comment]
Тут не всё так просто, как может казаться.

В идеале нужно сделать так, чтобы человек мог найти людей из нынешнего Первомайского на момент ПМВ, хотя тогда деревня называлась Лебедева.

Но поиск по координатам не подходит, т.к на практике были случаи, когда поселение переносили на значительные расстояния.

Надо думать...

**Comment by a-KaDeMi-k** [comment]
Описанный случай должен решаеться упомянутой мною доп. таблицей (связь многие ко многим).

Координаты - "да", не подходят. Один классический случай с Рязанью чего стоит!

**Comment by Limych** [comment]
Рязань - это ещё цветочки. На Кубани станица Ведмедевская из-за набегов горцев в XVIII в. была перенесена на несколько сотен км. А потом почти на том же месте появилась другая.

**Comment by a-KaDeMi-k** [comment]
Вот ещё вопрос, стоит ли считать такого рода "Рязани"  одним и тем же населённым пунктом...

**Comment by Limych** [comment]
Нет, конечно. Это два разных НП. Но вот Ведмедевская переехала со всем населением.  И это уже один НП, хоть и через пол Кубани.

**Comment by a-KaDeMi-k** [comment]
тоже спорно

**Comment by Limych** [comment]
Обоснуй :)

**Comment by a-KaDeMi-k** [comment]
Ну, можно это рассматривать как основание нового села в новом месте, но со старым названием. Как и Рязань.

Не утверждает же никто, что старая и новая Рязани - это один и тот же н.п. Даже уточняют, если надо - "старая" / "новая" или "современная"

Или вот ситуация с переселенцами, например, когда часть семей уходила в Сибирь и называла новые поселения в честь исходных родных сёл. 
Единственная разница с обсуждаемым примером - это что старое поселение продолжало существовать. Вроде как "продолжало". Сами переселенцы до поры до времени не могли с гарантией утверждать, что пока они переселялись с родиной ничего не произошло...

Так что на названия я бы ориентировался в гораздо меньшей степени, чем на координаты.
А так, с такого-то времени по такое-то в данном месте существовало поселение такое...
С такого-то по такое - ничего не было...
С такого-то по такое - было поселение с таким названием, а потом его переименовали в такое...
При этом с такого-то по такое - его могли одновременно называть ещё вот так-то...
